src/Server.js

```diff
@@ -204,7 +204,7 @@ export class Server extends EventEmitter {
 			}
 
 			// handle synchronous results
-			if (result && (!result.then || typeof result.then !== 'function')) {
+			if (typeof result !== 'undefined' && (!result.then || typeof result.then !== 'function')) {
```

Comment:
Same comment as above... `if (result)` will result in the same result as `if (typeof result !== 'undefined')`, but is more generic and a tad tighter in terms of implementation (plus more consistent with the rest of the code base). Is this a vestige from testing?

Reply:
Ah I think I see... the issue is that the result from the call is actually a scalar boolean value of `false`. In this case, your code change makes sense.